TransIT.BLL/Services/ImplementedServices/StatisticsService.cs

```diff
@@ -101,5 +102,76 @@ public async Task<List<int>> GetMalfunctionGroupStatistics(string malfunctionGro
 
             return result;
         }
+
+        public async Task<List<StatisticsDTO>> GetAllGroupsStatistics()
+        {
```

Comment:
try to rewrite all creating new lists logic to using iterator(yield), add 'Async' suffux to the method names that use async code

for ex:
` public async Task<IEnumerable<StatisticsDTO>> GetAllGroupsStatisticsAsync()
        {
            var malfunctionGroups = await _unitOfWork.MalfunctionGroupRepository.GetAllAsync();

             foreach (MalfunctionGroup group in malfunctionGroups)
            {
                yield return new StatisticsDTO
                {
                    FieldName = group.Name,
                    Statistics = await GetMalfunctionGroupStatistics(group.Name)
                };
            }
        }`

Reply:
As I found, to use this way we need to replace Task<IEnumerable<StatisticsDTO>> with IAsyncEnumerable<StatisticsDTO>, which now available only in "preview" language version. Is this ok?

Reply:
ok lets leave as is, I thought you have the latest SDK